packages/driver/src/options.ts

```diff
@@ -269,30 +271,36 @@ export class Options {
       clone.moduleAliases = this.moduleAliases;
     }
 
-    let originalCodecsMap = false;
     if (mergeOptions.codecs != null) {
       clone.codecs = new Map([
         ...this.codecs,
         ...Object.entries(mergeOptions.codecs),
       ]) as ReadonlyCodecMap;
     } else {
       clone.codecs = this.codecs;
-      originalCodecsMap = true;
+      // It  makes sense to just inherit the cached codec context
+      // as it's expensive to build. The only case when it can't be
+      // cached is when .withCodecs() was called, but caching will
+      // work great for all other .withSomething() methods we have.
+      clone.cachedCodecContext = this.cachedCodecContext;
+      clone.cachedCodecContextVer = this.cachedCodecContextVer;
```

Comment:
```
clone.cachedCodecContext = this.makeCodecContext();
```
If we want maximum caching™ we should make sure the `cachedCodecContext` is actually initalised before we copy it.

Reply:
There's a check when the cache is used, I think that check is enough.